xlb/operator/stepper/nse_stepper.py

```diff
@@ -91,92 +94,84 @@ def jax_implementation(self, f_0, f_1, bc_mask, missing_mask, timestep):
-                # Zouhe boundary condition (bc type = velocity)
-                f_post = self.ZouHeBC_velocity.warp_functional(index, timestep, missing_mask, f_0, f_1, f_pre, f_post)
-            elif _boundary_id == bc_struct.id_ZouHeBC_pressure:
-                # Zouhe boundary condition (bc type = pressure)
-                f_post = self.ZouHeBC_pressure.warp_functional(index, timestep, missing_mask, f_0, f_1, f_pre, f_post)
-            elif _boundary_id == bc_struct.id_RegularizedBC_velocity:
-                # Regularized boundary condition (bc type = velocity)
-                f_post = self.RegularizedBC_velocity.warp_functional(index, timestep, missing_mask, f_0, f_1, f_pre, f_post)
-            elif _boundary_id == bc_struct.id_RegularizedBC_pressure:
-                # Regularized boundary condition (bc type = velocity)
-                f_post = self.RegularizedBC_pressure.warp_functional(index, timestep, missing_mask, f_0, f_1, f_pre, f_post)
-            elif _boundary_id == bc_struct.id_ExtrapolationOutflowBC:
-                # Regularized boundary condition (bc type = velocity)
-                f_post = self.ExtrapolationOutflowBC.warp_functional(index, timestep, missing_mask, f_0, f_1, f_pre, f_post)
-            elif _boundary_id == bc_struct.id_GradsApproximationBC:
-                # Reformulated Grads boundary condition
-                f_post = self.GradsApproximationBC.warp_functional(index, timestep, missing_mask, f_0, f_1, f_pre, f_post)
-            return f_post
+            f_result = f_post
+
+            if wp.static("EquilibriumBC" in self.active_bcs):
+                if _boundary_id == wp.static(boundary_condition_registry.bc_to_id["EquilibriumBC"]):
+                    f_result = self.EquilibriumBC.warp_functional(index, timestep, missing_mask, f_0, f_1, f_pre, f_post)
+
+            if wp.static("DoNothingBC" in self.active_bcs):
+                if _boundary_id == wp.static(boundary_condition_registry.bc_to_id["DoNothingBC"]):
+                    f_result = self.DoNothingBC.warp_functional(index, timestep, missing_mask, f_0, f_1, f_pre, f_post)
+
+            if wp.static("HalfwayBounceBackBC" in self.active_bcs):
+                if _boundary_id == wp.static(boundary_condition_registry.bc_to_id["HalfwayBounceBackBC"]):
+                    f_result = self.HalfwayBounceBackBC.warp_functional(index, timestep, missing_mask, f_0, f_1, f_pre, f_post)
+
+            if wp.static("ZouHeBC" in self.active_bcs):
+                if _boundary_id == wp.static(boundary_condition_registry.bc_to_id["ZouHeBC"]):
+                    f_result = self.ZouHeBC_velocity.warp_functional(index, timestep, missing_mask, f_0, f_1, f_pre, f_post)
+
+            if wp.static("RegularizedBC" in self.active_bcs):
+                if _boundary_id == wp.static(boundary_condition_registry.bc_to_id["RegularizedBC"]):
+                    f_result = self.RegularizedBC.warp_functional(index, timestep, missing_mask, f_0, f_1, f_pre, f_post)
```

Comment:
These need to be broken into two if statements:
 ```
           if wp.static("RegularizedBC_velocity" in self.active_bcs):
                if _boundary_id == wp.static(boundary_condition_registry.bc_to_id["RegularizedBC_velocity"]):
                    f_result = self.RegularizedBC_velocity.warp_functional(index, timestep, missing_mask, f_0, f_1, f_pre, f_post)

            if wp.static("RegularizedBC_pressure" in self.active_bcs):
                if _boundary_id == wp.static(boundary_condition_registry.bc_to_id["RegularizedBC_pressure"]):
                    f_result = self.RegularizedBC_pressure.warp_functional(index, timestep, missing_mask, f_0, f_1, f_pre, f_post)
```

Reply:
Oh..This is actually a much bigger issue in BC. We can't have multiple type of the same BC with different prescribed values. Let me think about this more it needs more changes than this.